rpcserver.go

```diff
@@ -8134,7 +8134,12 @@ func (r *rpcServer) DeclareScriptKey(ctx context.Context,
 			err)
 	}
 
-	err = r.cfg.TapAddrBook.InsertScriptKey(ctx, *scriptKey, true)
+	// Because we've been given the key over the RPC interface, we can't be
+	// 100% sure of the type. But we can make a best effort guess based on
+	// the fields the user has set.
+	keyType := scriptKey.DetermineType()
```

Comment:
Any reason to not pass in the key type? That would give the caller a chance to do this determination.

Reply:
You mean from the RPC request? We'll do that in a later commit.